src/registry.js

```diff
@@ -1,24 +1,33 @@
 // @flow
-import type { Task, RegistredTopics, Runner } from '../types';
+import C from './constants';
+import type { Task, RegistredTopics, Runner, PublishCallback } from '../types';
 
-const Registry = (registeredTasks: RegistredTopics) => {
-  const addNewTask = (task: Task, runner: Runner) => {
+const Registry = (registeredTasks: RegistredTopics,
+      publishCallbacks: ?PublishCallback) => {
+  const addNewTask = async (task: Task, runner: Runner) => {
     registeredTasks[task.topic] = task; // eslint-disable-line
-    // call initialize consumer
-    const topics = Object.keys(registeredTasks);
-    runner.initializeConsumer(topics);
+    await runner.initializeGroupAdmin();
+    await runner.initializeProducer();
+    await runner.initializeConsumer(Object.keys(registeredTasks));
   };
 
-  const removeTask = (task: Task, runner: Runner) => {
+  const removeTask = async (task: Task, runner: Runner) => {
     delete registeredTasks[task.topic]; // eslint-disable-line
      // call initialize consumer
     const topics = Object.keys(registeredTasks);
-    runner.initializeConsumer(topics);
+    await runner.initializeProducer();
+    await runner.initializeGroupAdmin();
+    await runner.initializeConsumer(topics);
   };
 
   return {
     addNewTask,
     removeTask,
+    successCallback:
```

Comment:
i'm wondering if these should be more event based rather than callbacks, maybe your original idea of on a per task basis was better

`task.on('complete', () => {})`
`task.on('failure', () => {})`
`task.on('start', () => {})`